src/eko/constants.py

```diff
@@ -114,3 +117,22 @@ def charge_combinations(nf):
     vde2m = nd / nf * (eu2 - ed2)
     e2delta = vde2m - vue2m + e2avg
     return e2avg, vue2m, vde2m, e2delta
+
+
+@nb.njit(cache=True)
+def lepton_number(q2):
```

Comment:
maybe this is better located in `matchings.py` (since this is almost another atlas)

Reply:
I put it there since there was already `uplike_flavors`. But if you want I can move it

Reply:
I would still move it - as said it acts as an atlas (I was even briefly suggesting to do a full atlas, but then I thought this is too much); `uplike_flavors` instead is not an atlas and in fact unaware of any scale, but just does the "trivial" `nf // 2` - that is group theory and not physics